x-pack/plugins/security_solution/public/common/components/response_actions/endpoint_action_results.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiComment } from '@elastic/eui';
+import { FormattedRelative } from '@kbn/i18n-react';
+import React from 'react';
+import { useUserPrivileges } from '../user_privileges';
+import type { LogsEndpointAction } from '../../../../common/endpoint/types';
+import { useGetAutomatedActionResponseList } from '../../../management/hooks/response_actions/use_get_automated_action_list';
+import { ActionsLogExpandedTray } from '../../../management/components/endpoint_response_actions_list/components/action_log_expanded_tray';
+import type { ResponseActionsApiCommandNames } from '../../../../common/endpoint/service/response_actions/constants';
+import { ENDPOINT_COMMANDS } from '../event_details/translations';
+import { ResponseActionsEmptyPrompt } from './response_actions_empty_prompt';
+
+interface EndpointResponseActionResultsProps {
+  action: LogsEndpointAction;
+}
+
+export const EndpointResponseActionResults = ({ action }: EndpointResponseActionResultsProps) => {
+  const { rule, agent } = action;
+  const { action_id: actionId, expiration } = action.EndpointActions;
+  const {
+    endpointPrivileges: { canReadActionsLogManagement, canAccessEndpointActionsLogManagement },
+  } = useUserPrivileges();
+
+  const canReadEndpoint = canReadActionsLogManagement && canAccessEndpointActionsLogManagement;
+  const { data: responseData } = useGetAutomatedActionResponseList(
+    { actionId, expiration, agent },
+    { skip: !canReadEndpoint }
+  );
+
+  const eventText = getCommentText(action.EndpointActions.data.command);
```

Comment:
maybe we could use `useQuery().select` option to transform data to the expected shape, so we don't have to do it manually every time the component rerenders?

Reply:
Will try 👍